src/doc/trpl/dining-philosophers.md

```diff
@@ -2,45 +2,40 @@
-1965, but we’ll use the version from [this paper][paper] by Tony Hoare in 1985.
-
-[paper]: http://www.usingcsp.com/cspbook.pdf
-
-> In ancient times, a wealthy philanthropist endowed a College to accommodate
-> five eminent philosophers. Each philosopher had a room in which she could
-> engage in her professional activity of thinking; there was also a common
-> dining room, furnished with a circular table, surrounded by five chairs, each
-> labelled by the name of the philosopher who was to sit in it. They sat
-> anticlockwise around the table. To the left of each philosopher there was
-> laid a golden fork, and in the centre stood a large bowl of spaghetti, which
-> was constantly replenished. A philosopher was expected to spend most of her
-> time thinking; but when she felt hungry, she went to the dining room, sat down
-> in her own chair, picked up her own fork on her left, and plunged it into the
-> spaghetti. But such is the tangled nature of spaghetti that a second fork is
-> required to carry it to the mouth. The philosopher therefore had also to pick
-> up the fork on her right. When she was finished she would put down both her
-> forks, get up from her chair, and continue thinking. Of course, a fork can be
-> used by only one philosopher at a time. If the other philosopher wants it, she
-> just has to wait until the fork is available again.
+1965, and has been used as an example of the use of concurrency primitives in
+many influential works since then.
+
+Plato, Hypatia, Lao, Simone, and Audre are professors of philosophy at a
+university. As philosophers, they spend most of their time in their offices
+thinking and writing; but occasionally, they need to come out for food. There
+is a round table in the common room between their offices, and they each have
+a designated seat at it; the university ensures that there is always a fresh
+bowl of rice at each seat, so that they may come out and dine whenever they
+please. However, due to budgetary constraints, the university has only been
+able to afford five chopsticks for the five professors; in order to ensure
+that everyone can eat, the chopsticks are arranged between the places at the
+table, so each professor can pick up the two adjacent chopstics and begin
+eating. When each is done eating, she puts down her chopsticks in the
+original locations, and returns to her studies.
 
 This classic problem shows off a few different elements of concurrency. The
 reason is that it's actually slightly tricky to implement: a simple
 implementation can deadlock. For example, let's consider a simple algorithm
 that would solve this problem:
 
-1. A philosopher picks up the fork on their left.
-2. They then pick up the fork on their right.
-3. They eat.
-4. They return the forks.
+1. A philosopher picks up the chopstick on her left.
+2. She then picks up the chopstick on her right.
+3. She eats.
+4. She returns the chopsticks.
```

Comment:
Please be careful of the distinction between "unspecified" and "heterogenous" here. Chris is correct. You have started off your example with a group of _heterogenous_ (genderwise) philosophers (unlike the original example, which just mentioned philosophers). The genders are specified, just distinct. "she" is incorrect here, because some named male philosophers (Plato for example) are now labeled as women.

If they were unspecified (like the original example), it would be fine. The original example just talked of five philosophers, and later used "he" (which was changed to "she", which is also okay). This pronoun ends up specifying the genders.

Overwriting gender is incorrect. Specifying gender, when unspecified, is fine.

Fix: Introduce the philosophers later. Initially just go "Let us consider five philosophers".
Alternatively, rename all the philosophers to be female.